test/integration/ovs/openflow_test_utils.go

```diff
@@ -51,20 +51,27 @@ type ExpectFlow struct {
 	ActStr   string
 }
 
+func (f ExpectFlow) flowStr(id uint8) string {
+	return fmt.Sprintf("table=%d,%s actions=%s", id, f.MatchStr, f.ActStr)
+}
+
 func CheckFlowExists(t *testing.T, ovsCtlClient ovsctl.OVSCtlClient, tableID uint8, exist bool, flows []*ExpectFlow) []string {
 	flowList, _ := OfctlDumpTableFlows(ovsCtlClient, tableID)
 
+	failed := false
 	for _, flow := range flows {
 		found := OfctlFlowMatch(flowList, tableID, flow)
 		if exist && !found {
-			t.Errorf("Failed to install flow:\n%v", flow)
+			failed = true
+			t.Errorf("Failed to install flow: %s", flow.flowStr(tableID))
 		}
 		if !exist && found {
-			t.Errorf("Failed to uninstall flow:\n%v", flow)
+			failed = true
+			t.Errorf("Failed to uninstall flow: %s", flow.flowStr(tableID))
 		}
 	}
-	if t.Failed() {
-		t.Errorf("Existing flows:\n%v", flowList)
+	if failed {
+		t.Errorf("Existing flows:\n%s", strings.Join(flowList, "\n"))
```

Comment:
why this change?

Reply:
I think the change makes error logs more readable and neat.
* "Failed to install flow: *" should be one line.
* All existing flows are separated into several lines.

Before
```
--- FAIL: TestSNATFlows (1.19s)
    openflow_test_utils.go:60: Failed to install flow:
        &{priority=200,ct_state=+new+trk,ip,tun_dst=10.10.10.14 load:0xe->NXM_NX_PKT_MARK[0..7],goto_table:72}
    openflow_test_utils.go:60: Failed to install flow:
        &{priority=200,ct_state=+new+trk,ip,in_port=104 load:0xe->NXM_NX_PKT_MARK[0..7],goto_table:80}
    openflow_test_utils.go:60: Failed to install flow:
        &{priority=200,ip,in_port=204 set_field:aa:aa:aa:aa:aa:11->eth_src,set_field:aa:bb:cc:dd:ee:ff->eth_dst,set_field:10.10.10.14->tun_dst,goto_table:72}
    openflow_test_utils.go:67: Existing flows:
        [table=71, priority=200,ct_state=+new+trk,ip,tun_dst=10.10.10.14 actions=load:0xe->NXM_NX_PKT_MARK[0..7],goto_table:72 table=71, priority=200,ct_state=+new+trk,ipv6,tun_ipv6_dst=a963:ca9b:172:10::16 actions=load:0x10->NXM_NX_PKT_MARK[0..7],goto_table:72 table=71, priority=200,ct_state=+new+trk,ip,in_port=104 actions=load:0xe->NXM_NX_PKT_MARK[0..7],goto_table:80 table=71, priority=200,ct_state=+new+trk,ipv6,in_port=106 actions=load:0x10->NXM_NX_PKT_MARK[0..7],goto_table:80 table=71, priority=200,ip,in_port=204 actions=set_field:aa:aa:aa:aa:aa:11->eth_src,set_field:aa:bb:cc:dd:ee:ff->eth_dst,set_field:10.10.10.14->tun_dst,goto_table:72 table=71, priority=200,ipv6,in_port=206 actions=set_field:aa:aa:aa:aa:aa:11->eth_src,set_field:aa:bb:cc:dd:ee:ff->eth_dst,set_field:a963:ca9b:172:10::16->tun_ipv6_dst,goto_table:72 table=71, priority=0 actions=goto_table:80]
```

After
```
--- FAIL: TestSNATFlows (1.22s)
    openflow_test_utils.go:66: Failed to install flow: table=71,priority=200,ct_state=+new+trk,ip,tun_dst=10.10.10.14 actions=load:0xe->NXM_NX_PKT_MARK[0..7],goto_table:72
    openflow_test_utils.go:66: Failed to install flow: table=71,priority=200,ct_state=+new+trk,ip,in_port=104 actions=load:0xe->NXM_NX_PKT_MARK[0..7],goto_table:80
    openflow_test_utils.go:66: Failed to install flow: table=71,priority=200,ip,in_port=204 actions=set_field:aa:aa:aa:aa:aa:11->eth_src,set_field:aa:bb:cc:dd:ee:ff->eth_dst,set_field:10.10.10.14->tun_dst,goto_table:72
    openflow_test_utils.go:74: Existing flows:
        table=71, priority=200,ct_state=+new+trk,ip,tun_dst=10.10.10.14 actions=load:0xe->NXM_NX_PKT_MARK[0..7],goto_table:72
        table=71, priority=200,ct_state=+new+trk,ipv6,tun_ipv6_dst=a963:ca9b:172:10::16 actions=load:0x10->NXM_NX_PKT_MARK[0..7],goto_table:72
        table=71, priority=200,ct_state=+new+trk,ip,in_port=104 actions=load:0xe->NXM_NX_PKT_MARK[0..7],goto_table:80
        table=71, priority=200,ct_state=+new+trk,ipv6,in_port=106 actions=load:0x10->NXM_NX_PKT_MARK[0..7],goto_table:80
        table=71, priority=200,ip,in_port=204 actions=set_field:aa:aa:aa:aa:aa:11->eth_src,set_field:aa:bb:cc:dd:ee:ff->eth_dst,set_field:10.10.10.14->tun_dst,goto_table:72
        table=71, priority=200,ipv6,in_port=206 actions=set_field:aa:aa:aa:aa:aa:11->eth_src,set_field:aa:bb:cc:dd:ee:ff->eth_dst,set_field:a963:ca9b:172:10::16->tun_ipv6_dst,goto_table:72
        table=71, priority=0 actions=goto_table:80
```

Reply:
But it seems change to `t.Failed()` is unnecessary and the log would be the same. I don't understand why I made the change... Changed back.